Tools/Scripts/webkitpy/common/checkout/scm/git.py

```diff
@@ -548,22 +549,22 @@ def delete_branch(self, branch_name):
         if self.branch_ref_exists('refs/heads/' + branch_name):
             self._run_git(['branch', '-D', branch_name])
 
-    def remote_merge_base(self):
-        return self._run_git(['merge-base', self.remote_branch_ref(), 'HEAD']).strip()
-
-    def remote_branch_ref(self):
-        # Use references so that we can avoid collisions, e.g. we don't want to operate on refs/heads/trunk if it exists.
-        remote_branch_refs = self.read_git_config('svn-remote.svn.fetch', cwd=self.checkout_root, executive=self._executive)
-        if not remote_branch_refs:
-            for ref in ['refs/remotes/origin/main', 'refs/remotes/origin/master']:
-                if self.branch_ref_exists(ref):
-                    return ref
-            raise ScriptError(message="Can't find a branch to diff against. svn-remote.svn.fetch is not in the git config and neither main nor master exist")
-
-        # FIXME: What's the right behavior when there are multiple svn-remotes listed?
-        # For now, just use the first one.
-        first_remote_branch_ref = remote_branch_refs.split('\n')[0]
-        return first_remote_branch_ref.split(':')[1]
+    def remote_merge_base(self, find_branch=False):
+        return self._run_git(['merge-base', self.remote_branch_ref(find_branch), 'HEAD']).strip()
+
+    @memoized
+    def remote_branch_ref(self, find_branch=False):
+        if find_branch:
+            repository = local.Git(self.checkout_root)
+            branch_point = branch.Branch.branch_point(repository, limit=10)
+            if branch_point:
+                for remote in repository.source_remotes():
+                    if branch_point.branch in repository.branches_for(remote=remote, cached=True):
+                        return 'refs/remotes/{}/{}'.format(remote, branch_point.branch)
+        for ref in ['refs/remotes/origin/mainy', 'refs/remotes/origin/master']:
```

Comment:
typo mainy -> main

Reply:
Oh oops, this was for testing anyways, thanks for catching that.